src/apps/jsgeneric/jsgeneric.cpp

```diff
@@ -13,33 +13,101 @@ namespace ccfapp
   using namespace kv;
   using namespace ccf;
 
-  using GenericTable = ccf::Store::Map<nlohmann::json, nlohmann::json>;
+  using LogTable = ccf::Store::Map<int32_t, std::string>;
+
+  static JSValue js_print(
+    JSContext* ctx, JSValueConst this_val, int argc, JSValueConst* argv)
+  {
+    int i;
+    const char* str;
+    std::stringstream ss;
+
+    for (i = 0; i < argc; i++)
+    {
+      if (i != 0)
+        ss << ' ';
+      str = JS_ToCString(ctx, argv[i]);
+      if (!str)
+        return JS_EXCEPTION;
+      ss << str;
+      JS_FreeCString(ctx, str);
+    }
+    LOG_INFO_FMT(ss.str());
+    return JS_UNDEFINED;
+  }
+
+  void js_dump_error(JSContext* ctx)
+  {
+    JSValue exception_val = JS_GetException(ctx);
+
+    JSValue val;
+    const char* stack;
+    bool is_error;
+
+    is_error = JS_IsError(ctx, exception_val);
+    if (!is_error)
+      LOG_INFO_FMT("Throw: ");
```

Comment:
Minor: should these be `LOG_FAIL_FMT()`?

Reply:
No, FAIL is an important failure inside CCF, this is just business logic raising an exception. It's perfectly safe, handled etc. In fact, in production code, this should be turned off, because it can leak information about what the business logic is doing.